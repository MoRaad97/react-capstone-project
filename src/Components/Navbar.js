import React from "react";
import './style/Navbar.css'

const Navbar = (props) => {
  return (
    <nav className={props.theme === "default" ? 'navbar df-nav' : 'navbar'}>
      <div className="container">
        <h1>Where in the world?</h1>
        {props.theme === "default" && <h2 onClick={props.themeSwitch}>White Theme</h2>}
        {props.theme === "white" && <h2 onClick={props.themeSwitch}>Default Theme</h2>}
      </div>

    </nav>
  )
}
export default Navbar