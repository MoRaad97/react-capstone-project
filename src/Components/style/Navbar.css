.navbar {
  padding: 25px 0;
}

.df-nav {
  padding: 25px 0;
  background-color: #ec4c8a;
  color: white;
}

.navbar > .container {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.navbar h1 {
  margin: 0;
  font-size: 20px;
  text-transform: capitalize;
  font-family: "Gill Sans", sans-serif;
  font-weight: 900;
}

.navbar > .container > div {
  margin: 0;
  font-size: 18px;
  font-weight: 700;
  text-transform: capitalize;
  font-family: "Lato", sans-serif;
}
