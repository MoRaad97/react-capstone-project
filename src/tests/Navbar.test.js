import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/Store';
import Navbar from '../Components/Navbar';

it('Navbar Component snapshot test', () => {
  const NavbarRender = TestRenderer
    .create(
      <Provider store={store}>
        <Navbar />
      </Provider>,
    )
    .toJSON();
  expect(NavbarRender).toMatchSnapshot();
});