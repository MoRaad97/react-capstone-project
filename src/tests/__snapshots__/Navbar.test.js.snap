// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component snapshot test 1`] = `
<nav
  className="navbar"
>
  <div
    className="container"
  >
    <h1>
      Where in the world?
    </h1>
  </div>
</nav>
`;
